include/seastar/core/chunked_fifo.hh

```diff
@@ -142,6 +142,7 @@ private:
         size_t _item_index = 0;
 
     protected:
+        inline basic_iterator() noexcept;
```

Comment:
Why this change? It looks like `iterator` and `const_iterator` both already have default constructors.

Reply:
Because of https://github.com/scylladb/seastar/blob/master/src/core/scollectd.cc#L415-L418

Clang was complaining that chunked_fifo did not have a default constructor for that call to create a `shared_ptr<context>()`

Reply:
Ah right and `iterator` and `const_iterator` are marked as `default`.